src/Elliptic/Executables/NonlinearEllipticSolver.hpp

```diff
@@ -0,0 +1,245 @@
+#include "Options/Options.hpp"
+#include "ParallelAlgorithms/Actions/AddComputeTags.hpp"
+#include "ParallelAlgorithms/LinearSolver/Actions/MakeIdentityIfSkipped.hpp"
+#include "ParallelAlgorithms/LinearSolver/Gmres/Gmres.hpp"
+#include "ParallelAlgorithms/LinearSolver/Multigrid/Actions/RestrictFields.hpp"
+#include "ParallelAlgorithms/LinearSolver/Multigrid/Multigrid.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Actions/CommunicateOverlapFields.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Actions/ResetSubdomainSolver.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Schwarz.hpp"
+#include "ParallelAlgorithms/LinearSolver/Tags.hpp"
+#include "ParallelAlgorithms/NonlinearSolver/NewtonRaphson/NewtonRaphson.hpp"
+#include "ParallelAlgorithms/NonlinearSolver/Tags.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/AnalyticSolution.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/Background.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialGuess.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace elliptic::nonlin_solver {
+
+namespace OptionTags {
+
+struct NonlinearSolverGroup {
+  static std::string name() { return "NonlinearSolver"; }
+  static constexpr Options::String help = "The iterative nonlinear solver";
+};
+struct NewtonRaphsonGroup {
+  static std::string name() { return "NewtonRaphson"; }
+  static constexpr Options::String help =
+      "Options for the Newton-Raphson nonlinear solver";
+  using group = NonlinearSolverGroup;
+};
+struct LinearSolverGroup {
+  static std::string name() { return "LinearSolver"; }
+  static constexpr Options::String help =
+      "The iterative Krylov-subspace linear solver";
+};
+struct GmresGroup {
+  static std::string name() { return "Gmres"; }
+  static constexpr Options::String help = "Options for the GMRES linear solver";
+  using group = LinearSolverGroup;
+};
+struct SchwarzSmootherGroup {
+  static std::string name() { return "SchwarzSmoother"; }
+  static constexpr Options::String help = "Options for the Schwarz smoother";
+  using group = LinearSolverGroup;
+};
+struct MultigridGroup {
+  static std::string name() { return "Multigrid"; }
+  static constexpr Options::String help = "Options for the multigrid";
+  using group = LinearSolverGroup;
+};
+
+}  // namespace OptionTags
+
+/*!
+ * \brief A complete nonlinear elliptic solver stack. Use to compose an
+ * executable.
+ */
+template <typename Metavariables>
+struct Solver {
+  static constexpr size_t volume_dim = Metavariables::volume_dim;
+  using system = typename Metavariables::system;
+  static_assert(
+      tt::assert_conforms_to_v<system, elliptic::protocols::FirstOrderSystem>);
+
+  using background_tag =
+      elliptic::Tags::Background<elliptic::analytic_data::Background>;
+  using initial_guess_tag =
+      elliptic::Tags::InitialGuess<elliptic::analytic_data::InitialGuess>;
+
+  // These are the fields we solve for
```

Comment:
`dox` the type aliases? You dox the class, so might be nice to have the member type aliases that you've deemed worth commenting be documented in doxygen